lib/msal-node/test/client/ManagedIdentityApplication.spec.ts

```diff
@@ -0,0 +1,326 @@
+ */
+
+import { ManagedIdentityApplication } from "../../src/client/ManagedIdentityApplication";
+import { ManagedIdentityConfiguration } from "../../src/config/Configuration";
+import { ManagedIdentityRequestParams } from "../../src/request/ManagedIdentityRequestParams";
+import {
+    DEFAULT_MANAGED_IDENTITY_AUTHENTICATION_RESULT,
+    MANAGED_IDENTITY_CLIENT_ID,
+    MANAGED_IDENTITY_RESOURCE,
+} from "../test_kit/StringConstants";
+
+import { ManagedIdentityTestUtils } from "../test_kit/ManagedIdentityTestUtils";
+import {
+    DEFAULT_MANAGED_IDENTITY_ID,
+    ManagedIdentityIdType,
+} from "../../src/utils/Constants";
+
+describe("ManagedIdentityApplication unit tests", () => {
+    let OLD_ENVS: NodeJS.ProcessEnv;
+    beforeAll(() => {
+        // make a copy of old environment
+        OLD_ENVS = process.env; // Restore old environment
+    });
+    afterAll(() => {
+        // restore old environment
+        process.env = OLD_ENVS;
+    });
+
+    const managedIdentityRequestParams: ManagedIdentityRequestParams = {
+        resource: MANAGED_IDENTITY_RESOURCE,
+    };
+
+    const nonSystemAssignedConfig: ManagedIdentityConfiguration = {
+        system: {
+            networkClient:
+                ManagedIdentityTestUtils.getManagedIdentityNetworkClient(
+                    MANAGED_IDENTITY_CLIENT_ID
+                ),
+        },
+        managedIdentityIdParams: {
+            userAssignedClientId: MANAGED_IDENTITY_CLIENT_ID,
+            // or
+            // userAssignedResourceId?: string;
+            // or
+            // userAssignedObjectId?: string;
+        },
+    };
+
+    const systemAssignedConfig: ManagedIdentityConfiguration = {
+        system: {
+            networkClient:
+                ManagedIdentityTestUtils.getManagedIdentityNetworkClient(
+                    DEFAULT_MANAGED_IDENTITY_ID
+                ),
+            // managedIdentityIdParams will be omitted for system-assigned
+        },
+    };
+
+    describe("Constructor", () => {
+        it("creates a ClientCredentialClient", async () => {
+            const managedIdentityApplication = new ManagedIdentityApplication(
+                systemAssignedConfig
+            );
+            expect(managedIdentityApplication).not.toBeNull();
+            expect(
+                managedIdentityApplication instanceof ManagedIdentityApplication
+            ).toBe(true);
+        });
+    });
+
+    describe("Acquires a token successfully via an App Service Managed Identity", () => {
+        beforeAll(() => {
+            // make a copy
+            process.env = { ...OLD_ENVS };
+
+            process.env["IDENTITY_ENDPOINT"] = "fake_IDENTITY_ENDPOINT";
+            process.env["IDENTITY_HEADER"] = "fake_IDENTITY_HEADER";
+        });
+
+        afterAll(() => {
+            // restore old environment
+            process.env = OLD_ENVS;
+        });
+
+        test("acquires a Non-System-Assigned token", async () => {
```

Comment:
Pls avoid calling it "non-system-assigned" and use the term "user assigned" when you mean user assigned + client ID and "user assigned by object ID" etc.